agent/consul/state/config_entry.go

```diff
@@ -1435,6 +1441,28 @@ func readDiscoveryChainConfigEntriesTxn(
 		res.Services[svcID] = entry
 	}
 
+	peerEntMeta := structs.DefaultEnterpriseMetaInPartition(entMeta.PartitionOrDefault())
+	for peerName := range todoPeers {
+		q := Query{
+			Value:          strings.ToLower(peerName),
```

Comment:
Nit. This can probably be left as-is, but peer names are required to be lowercase now.